capellambse/svg/drawing.py

```diff
@@ -451,38 +423,49 @@ def draw_object(self, obj: cabc.Mapping[str, t.Any]) -> None:
         obj
             The (decoded) JSON-dict of a single diagram object.
         """
+        mobj = copy.deepcopy(obj)
+        del obj
+        type_mapping: dict[str, tuple[t.Any, str]] = {
+            "box": (self._draw_box, "Box"),
+            "edge": (self._draw_edge, "Edge"),
+            "circle": (self._draw_circle, "Edge"),
+            "symbol": (self._draw_symbol, "Box"),
+            "box_symbol": (self._draw_box_symbol, "Box"),
+        }
+
         try:
-            drawfunc: t.Any = getattr(self, f'_draw_{obj["type"]}')
-        except AttributeError:
-            raise ValueError(f'Invalid object type: {obj["type"]}') from None
+            drawfunc, style_type = type_mapping[mobj["type"]]
+        except KeyError:
+            raise ValueError(f"Invalid object type: {mobj['type']}") from None
 
-        mobj = copy.deepcopy(obj)
         objparams = {
-            f"{k}_": v for k, v in obj.items() if k not in {"type", "style"}
+            f"{k}_": v for k, v in mobj.items() if k not in {"type", "style"}
         }
-        if obj["class"] in decorations.all_ports:
+        if mobj["class"] in decorations.all_ports:
             mobj["type"] = "box"  # type: ignore[index]
 
-        class_: str = mobj["type"].capitalize() + (
-            f".{mobj['class']}" if "class" in obj else ""
+        class_: str = style_type + (
+            f".{mobj['class']}" if "class" in mobj else ""
         )
+        my_styles: dict[str, t.Any] = {
+            **capstyle.get_style(self.diagram_class, class_),
+            **mobj.get("style", {}),
+        }
         obj_style = style.Styling(
             self.diagram_class,
             class_,
-            **{k: v for k, v in obj.get("style", {}).items() if "_" not in k},
+            **{k: v for k, v in my_styles.items() if "_" not in k},
         )
         text_style = style.Styling(
             self.diagram_class,
             class_,
             "text",
             **{
-                k[len("text_") :]: v
-                for k, v in obj.get("style", {}).items()
-                if k.startswith("text_")
+                k[5:]: v for k, v in my_styles.items() if k.startswith("text_")
             },
         )
 
-        self.obj_cache[mobj["id"]] = obj
+        self.obj_cache[mobj["id"]] = mobj
```

Comment:
This `self.obj_cache` seems to no longer be used at all. It's only written to, but never read anywhere. While we're cleaning up here anyways, might as well remove that too.

Reply:
I found that `self.obj_cache` is used in the function `get_port_transformation` on line 341. When you remove that part of the code the ports don't rotate as they should on the diagram.

Should we find a work around for this?

Reply:
Ah, interesting. I didn't find that with my cursory glance. Let's keep it in, that gets out of scope otherwise.